### ISSUE TITLE
FlopCounterMode counts 0 FLOPs for CPU scaled_dot_product_attention (flop_registry lacks aten::_scaled_dot_product_flash_attention_for_cpu)

### ISSUE THREAD

**Issue by johnmarktaylor91** [pytorch/pytorch#195460]
### 🐛 Describe the bug

`torch.utils.flop_counter.FlopCounterMode` reports **0 FLOPs** for `scaled_dot_product_attention` on CPU, because the CPU default path dispatches to `aten::_scaled_dot_product_flash_attention_for_cpu`, which has no entry in `flop_registry`. The CUDA counterpart (`aten::_scaled_dot_product_flash_attention`) is registered, so the same model reports correct attention FLOPs on GPU and silently drops them on CPU. Note this affects eval AND train mode: with dropout 0.0 the CPU dispatcher still selects the fused `_for_cpu` kernel.

Repro (torch 2.13.0):

```python
import torch
from torch.utils.flop_counter import FlopCounterMode, flop_registry

q = torch.randn(1, 12, 64, 64)  # (B, H, S, D)
with FlopCounterMode(display=False) as m:
    torch.nn.functional.scaled_dot_product_attention(q, q, q)
print(m.get_total_flops())  # 0  (expected: 12_582_912)

print(torch.ops.aten._scaled_dot_product_flash_attention_for_cpu in flop_registry)  # False
print(torch.ops.aten._scaled_dot_product_flash_attention in flop_registry)          # True
```

Dispatch confirmation: a `TorchDispatchMode` spy on the call above sees `aten._scaled_dot_product_flash_attention_for_cpu.default`.

Expected count for the shape above: QK^T + AV = 2 x (2 x B x H x S^2 x D) = 12,582,912 FLOPs per call (150,994,944 across a 12-layer stack), matching what `sdpa_flop` computes for the registered CUDA overloads.

Suggested fix: register the `_for_cpu` overload (and its backward, `aten::_scaled_dot_product_flash_attention_for_cpu_backward`, alongside the existing `sdpa_backward` registrations) with the existing `sdpa_flop` / `sdpa_backward_flop` formulas in `torch/utils/flop_counter.py`.

Found while cross-checking FLOP counts against dispatcher-level op logs in [torchlens](https://github.com/johnmarktaylor91/torchlens); happy to send a PR if useful.

### Versions

Reproduced on torch 2.13.0 (cu130 build, CPU execution). The `_for_cpu` overload is absent from `flop_registry` on every version we checked back to 2.1.
